<?xml version="1.0"?>
<launch>
  <!-- Navigation Controller -->
  <node pkg="navigation_pkg" type="navigation_controller_node" name="navigation_controller" output="screen">
    <param name="waypoint_reached_threshold" value="0.5" />
    <param name="max_velocity" value="2.0" />
  </node>

  <!-- Landing Controller -->
  <node pkg="navigation_pkg" type="landing_controller_node" name="landing_controller" output="screen">
    <param name="takeoff_height" value="2.0" />
    <param name="hover_height" value="2.0" />
    <param name="takeoff_velocity" value="0.5" />
    <param name="landing_velocity" value="0.3" />
    <param name="hover_duration" value="3.0" />
    <param name="control_frequency" value="20.0" />
    <param name="ground_threshold" value="0.1" />
    <param name="stability_threshold" value="0.1" />
  </node>
  
  <!-- Trajectory Tracker -->
  <node pkg="trajectory_tracking_pkg" type="trajectory_tracker_node" name="trajectory_tracker" output="screen">
    <param name="tracking_frequency" value="50.0" />
    <param name="look_ahead_time" value="0.2" />
    <param name="timeout_duration" value="5.0" />
  </node>
</launch>